opte/src/engine/packet.rs

```diff
@@ -1498,7 +1548,19 @@ impl Packet<Parsed> {
             }
 
             Some(IpMeta::Ip6(ip6)) => {
-                ip6.pay_len = (new_pkt_len - pkt_offset) as u16;
+                // IPv6 Payload Length field is defined in RFC 2640 section 3
+                // as:
+                //
+                // > Length of the IPv6 payload, i.e., the rest of the packet
+                // > following this IPv6 header, in octets. (Note that any
+                // > extension headers [section 4] present are considered part
+                // > of the payload, i.e., included in the length count.)
+                //
+                // So we need to remove the size of the fixed header (40
+                // octets), which is included in the total new packet length,
+                // when setting the payload length.
+                ip6.pay_len =
+                    (new_pkt_len - pkt_offset - Ipv6Hdr::BASE_SIZE) as u16;
```

Comment:
I thought this looked familiar! I'm confused, I see this exact diff [here](https://github.com/oxidecomputer/opte/pull/340/commits/ff9873b4fba809c7f8817ff336938ece89c35f03) from #340.

Reply:
Right, that PR fixed the calculation on the outer header (`Packet::emit_outer_headers`) but not the inner header (`Packet::emit_inner_headers`).